product_expiry_assign/models/stock_move.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2025 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import api, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def with_context(self, *args, **kwargs) -> api.Self:
```

Comment:
I'd rather add an hook there instead of doing this hack. Any argument against?

Reply:
I changed the implementation to instead catch the `with_expiration` ctx key where it's used: in `<stock.quant>._get_gather_domain(...)` (see fixup). Does it sound better?